source/extensions/access_loggers/wasm/config.cc

```diff
@@ -24,16 +24,12 @@ WasmAccessLogFactory::createAccessLogInstance(const Protobuf::Message& proto_con
       const envoy::extensions::access_loggers::wasm::v3::WasmAccessLog&>(
       proto_config, context.messageValidationVisitor());
 
-  // Create a base WASM to verify that the code loads before setting/cloning the for the
-  // individual threads.
   auto plugin = std::make_shared<Common::Wasm::Plugin>(
-      config.config().name(), config.config().root_id(), config.config().vm_config().vm_id(),
-      config.config().vm_config().runtime(),
-      Common::Wasm::anyToBytes(config.config().configuration()), config.config().fail_open(),
-      envoy::config::core::v3::TrafficDirection::UNSPECIFIED, context.localInfo(),
+      config.config(), envoy::config::core::v3::TrafficDirection::UNSPECIFIED, context.localInfo(),
       nullptr /* listener_metadata */);
 
-  auto access_log = std::make_shared<WasmAccessLog>(plugin, nullptr, std::move(filter));
+  auto access_log =
+      std::make_shared<WasmAccessLog>(config.config(), plugin, nullptr, std::move(filter));
```

Comment:
Couldn't `WasmBaseConfig` be a member of `Plugin` instead?

Reply:
nice, that would be better. Will do.